### PR TITLE
refactor: rewrite forge doc server using axum

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

We're already using `axum` for Anvil, let's not use another one for no reason.

## Solution

- Rewrite the simple HTTP server with axum and tower-http.
- The WebSocket autoreload stuff did not do anything, so I removed it.
- Move it to `forge` binary crate since it's pretty heavy dependency-wise.
- While I'm at it, added a `--open` to open the docs in a browser after serving them.

### CODE REVIEW

File: crates/forge/bin/cmd/doc/server.rs [hunk 1]
```diff
@@ -0,0 +1,104 @@
+use axum::{routing::get_service, Router};
+use forge_doc::mdbook::{utils::fs::get_404_output_file, MDBook};
+use std::{
+    net::{SocketAddr, ToSocketAddrs},
+    path::PathBuf,
+};
+use tower_http::services::{ServeDir, ServeFile};
+
+/// The HTTP endpoint for the websocket used to trigger reloads when a file changes.
+const LIVE_RELOAD_ENDPOINT: &str = "/__livereload";
```
Comment (mattsse) [inline crates/forge/bin/cmd/doc/server.rs:10]: oh hell yeah,
I think for reloading we'd need a watch service that runs forge doc?

File: crates/forge/bin/cmd/doc/mod.rs [hunk 2]
```diff
@@ -33,6 +36,10 @@ pub struct DocArgs {
     #[clap(long, short)]
     serve: bool,
 
+    /// Open the documentation in a browser after serving.
+    #[clap(long, requires = "serve")]
+    open: bool,
```
Comment (mattsse) [inline crates/forge/bin/cmd/doc/mod.rs:41]: amazing, 
works

**Review by mattsse — APPROVED** [review]
lgtm,
no need for warp, yes

for hotreloading, we could install watchexec with -w option

File: crates/forge/bin/cmd/doc/server.rs (hunk 1 above)
Comment (DaniPopes) [reply crates/forge/bin/cmd/doc/server.rs:10]: This was half-implemented. I can open a follow up implementing this

Comment (mattsse) [reply crates/forge/bin/cmd/doc/server.rs:10]: sweet,

I can't really remember how the watchexec API works, but I believe it was registering an action via a closure, maybe the forge watch command is helpful, perhaps not

**Review by onbjerg — APPROVED** [review]

**Pull request merged**